File: snippets/csharp/System.Windows.Forms/TabControl/ImageList/form1.cs
﻿// <snippet1>
using System.Drawing;
using System.ComponentModel;
using System.Windows.Forms;
using System.Resources;

public class Form1 : Form
{
    public Form1()
    {
        IContainer components = new Container();
        ResourceManager resources = new ResourceManager(typeof(Form1));
        TabControl tabControl1 = new TabControl();
        TabPage tabPage1 = new TabPage();

        // Declares and instantiates the ImageList object.
        ImageList myImages = new ImageList(components);

        tabControl1.Controls.Add(tabPage1);
        // Sets the images in myImages to display on the tabs of tabControl1. 
        tabControl1.ImageList = myImages;
 
        tabPage1.ImageIndex = 0;
        tabPage1.Text = "tabPage1";

        // Gets the handle that provides the data of myImages.
        myImages.ImageStream = ((ImageListStreamer)(resources.GetObject("myImages.ImageStream")));
    
        // Sets properties of myImages. 
        myImages.ColorDepth = ColorDepth.Depth8Bit;
        myImages.ImageSize = new Size(16, 16);
        myImages.TransparentColor = Color.Transparent;

        this.Controls.Add(tabControl1);
    }

    static void Main() 
    {
        Application.Run(new Form1());
    }
}
// </snippet1>

File: snippets/visualbasic/VS_Snippets_Winforms/TabPage.ImageIndex/VB/form1.vb
﻿' <snippet1>
Imports System.Drawing
Imports System.ComponentModel
Imports System.Windows.Forms
Imports System.Resources

Public Class Form1
    Inherits Form

    Public Sub New()
        Dim components = New Container()
        Dim resources As New ResourceManager(GetType(Form1))
        Dim tabControl1 As New TabControl()
        Dim tabPage1 As New TabPage()
        Dim myImages As New ImageList(components)

        tabControl1.Controls.Add(tabPage1)

        ' Displays images from myImages on the tabs of tabControl1.
        tabControl1.ImageList = myImages

        ' Specifies which image to display (on the tab of tabPage1) by its index.
        tabPage1.ImageIndex = 0

        tabPage1.Text = "tabPage1"

        myImages.ImageStream = CType(resources.GetObject("myImages.ImageStream"), ImageListStreamer)
        myImages.ColorDepth = ColorDepth.Depth8Bit
        myImages.ImageSize = New Size(16, 16)
        myImages.TransparentColor = Color.Transparent

        Me.Controls.Add(tabControl1)
    End Sub

    Shared Sub Main()
        Application.Run(New Form1())
    End Sub
End Class
' </snippet1>
